llm_toolkit/models.py

```diff
@@ -204,7 +204,7 @@ def estimate_token_num(self, text) -> int:
     """Estimates the number of tokens in |text|."""
     # https://cookbook.openai.com/examples/how_to_count_tokens_with_tiktoken
     try:
-      encoder = tiktoken.encoding_for_model(self.name)
+      encoder = tiktoken.encoding_for_model(self.name.replace('-azure', ''))
```

Comment:
I missed this one before, I thought it was only the client that changed.

This looks good but I have one more suggestion to improve readability and allow for potential future extensions: We should add two methods: `_get_tiktoken_encoding` and `_get_client` to this class. Here they would return the GPT values and then override them in `AzureGPT` below to return the Azure versions.

Reply:
That's ok, I have finished the update. :)  @mihaimaruseac

Reply:
Also, I can help to update the `README` and `Usage` doc for the Azure support & usage if the above code looks good. :)

Reply:
> Also, I can help to update the `README` and `Usage` doc for the Azure support & usage if the above code looks good. :)

Hi @mihaimaruseac . Do I need to update these two files in this PR? If so, I can also do the update below at the same time:
> Very minor nit: We can also make the method receive as argument the name of the model (in the parent class), then this one would just be
`super()._get_tiktoken_encoding(self.name.replace('-azure', ''))`
and the rest of the lines would be handled by the parent class.

Reply:
Sorry, I forgot to trigger some of the CI and was waiting on it to complete before merging. This is ready to merge, I'll merge it by the end of today.

Please feel free to do the README/Usage and `super` changes in a new PR, or in this one. It's up to you